pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -190,20 +187,10 @@ let
                      optionals doCheck (testDepends ++ testHaskellDepends ++ testSystemDepends ++ testFrameworkDepends) ++
                      optionals doBenchmark (benchmarkDepends ++ benchmarkHaskellDepends ++ benchmarkSystemDepends ++ benchmarkFrameworkDepends);
 
-  allBuildInputs = propagatedBuildInputs ++ otherBuildInputs;
-
-  haskellBuildInputs = stdenv.lib.filter isHaskellPkg allBuildInputs;
-  systemBuildInputs = stdenv.lib.filter isSystemPkg allBuildInputs;
-
-  # When not cross compiling, also include Setup.hs dependencies.
-  ghcEnv = ghc.withPackages (p:
-    haskellBuildInputs ++ stdenv.lib.optional (!isCross) setupHaskellDepends);
```

Comment:
A notable difference with how it's now is that setupHaskellDepends are always included in the shell. It now uses this here to determine which haskell packages to include: https://github.com/NixOS/nixpkgs/blob/463704a62e3564b951eff84909eeffea4d728130/pkgs/development/haskell-modules/lib.nix#L358-L383

Reply:
That's probably a good change, IMO

Reply:
Now that code is deleted, so where does it happen now?

Reply:
@Ericson2314 In `shellFor`.